src/util/crypto/libcrypto/crypto_sha512crypt.c

```diff
@@ -317,9 +317,14 @@ static int sha512_crypt_r(const char *key,
      * as well.  */
     EVP_MD_CTX_free(ctx);
     EVP_MD_CTX_free(alt_ctx);
-    if (p_bytes) memset(p_bytes, '\0', key_len);
-    if (s_bytes) memset(s_bytes, '\0', salt_len);
-    memset(temp_result, '\0', sizeof(temp_result));
+    if (p_bytes != NULL) {
+        sss_erase_mem_securely(p_bytes, key_len);
+    }
+    if (s_bytes != NULL) {
+        sss_erase_mem_securely(s_bytes, key_len);
```

Comment:
`salt_len` instead of `key_len` should be used here.

Reply:
Good catch! Thank you.